Rewrite LiveSearch as an ES Module

kr8n3r opened the pull request:
We want to migrate all of our legacy Javascript files to ESM, so this is the latest in the series
of conversions.

Rewrite follows the same conventions we previously used for FocusBanner, Collapsible checkboxes,
ColourPreview, FileUpload, Autofocus, PreviewPane and CopyToClipboard.

Added bonus: removes the use of jQuery.

In this PR we:

- rewrite the module as ES Module
- update the tests to be ES Module
- remove old module JS file
- remove old module from the legacy bundle build

## How to test

- templates page search
- team members search
- change branding search
- /organisations/<id>/settings/email-branding/add
-  /organisations/<id>/settings/letter-branding/add
-  /organisations/<id>/settings/add-nhs-local-organisation for NHS services
-  /services/<id>/service-settings/link-service-to-organisation
-  /services/>id>/service-settings/(set-email/letter)-branding

kr8n3r committed:
Update tests

rename and adapt to ES Module

kr8n3r committed:
Remove old LiveSearch JS from the bundle

kr8n3r committed:
Rewrite LiveSearch as an ES Module

We want to migrate all of our legacy Javascript
files to ESM, so this is the latest in the series
of conversions.

Rewrite follows the same conventions we previously
used for FocusBanner, Collapsible checkboxes,
ColourPreview, FileUpload, Autofocus,
PreviewPane and CopyToClipboard.

Added bonus: removes the use of jQuery as well

tombye reviewed (changes requested):
This looks good to me. I spotted a few things, one a bit pedantic and the other more of a question. Everything else great though 👍🏻

kr8n3r committed:
Add additions styles for elements with "hidden" attribute

We use hidden attribute to hide items that don't match live-search results.

Normally, the broser would hide the element, but as these ones have the display property set, we need to set additional override styles.

kr8n3r commented on app/assets/javascripts/esm/live-search.mjs:
```diff
@@ -0,0 +1,87 @@
+    let query = this.normalize(this.$searchBox.value);
+    let results = 0;
+
+    $targets.forEach((node) => {
```
agreed, fixed

kr8n3r commented on app/assets/javascripts/esm/live-search.mjs:
```diff
@@ -0,0 +1,87 @@
+      let isMatch = this.normalize(content).includes(this.normalize(query));
+      // if there is a child node with checked state
+      if (node.querySelectorAll(':checked').length > 0) {
+        node.style.display = 'block';
```
yep, made the changes. 
Additonally,some CSS needed to be set. See https://github.com/alphagov/notifications-admin/pull/5370/commits/c036ea6432df096637a5664fadaff8a43c37a935

kr8n3r commented:
spotted an issue with template search :(

kr8n3r committed:
Add more logic to template list visibility

There are some items that are hidden by default with CSS and which we need to show if they match the search query.

Whiel we do the visibility toggling by applying the `hidden` attribute to the each item, we cannot use the same attribute of the hiden-by-default items, as the checkbox macro from DS does not accomodate for that, hence they're hidden by css class.

This adds a bit more logic to the JS to apply a new class to those items when they match the search query and use CSS to show them, as well as remove the same class when they don't match or the search query is empty (default state)

kr8n3r committed:
Extra class to show item's hidden by default

When they match the search query

kr8n3r commented:
> spotted an issue with template search :(

issue resolved with 
https://github.com/alphagov/notifications-admin/pull/5370/commits/70b2fde55130e62f63a174a9313b02ff45c846b7
https://github.com/alphagov/notifications-admin/pull/5370/commits/8966f33a840549ffb7c77a518833eafe4acbe7de

tombye reviewed (approved):
Sorry for the late re-review, all looking good now 👍🏻
